scanpy/tests/test_highly_variable_genes.py

```diff
@@ -514,3 +514,39 @@ def test_cellranger_n_top_genes_warning():
         match="`n_top_genes` > number of normalized dispersions, returning all genes with normalized dispersions.",
     ):
         sc.pp.highly_variable_genes(adata, n_top_genes=1000, flavor="cell_ranger")
+
+
+@pytest.mark.parametrize("flavor", ["seurat", "cell_ranger"])
+@pytest.mark.parametrize("subset", [True, False])
+@pytest.mark.parametrize("inplace", [True, False])
+def test_highly_variable_genes_subset_inplace_consistency(
+    flavor,
+    subset,
+    inplace,
+):
+    if flavor == "seurat" or flavor == "cell_ranger":
+        adata = pbmc3k()
```

Comment:
Since for this test shape is most relevant, will try to use a minimalistic dataset (which does not clash with the highly_variable_genes flavors)